libraries/AP_BattMonitor/AP_BattMonitor.cpp

```diff
@@ -243,12 +251,8 @@ bool AP_BattMonitor::is_powering_off(uint8_t instance) const {
 /// has_current - returns true if battery monitor instance provides current info
 bool AP_BattMonitor::has_current(uint8_t instance) const
 {
-    // check for analog voltage and current monitor or smbus monitor
-    if (instance < _num_instances && drivers[instance] != nullptr) {
-        return (_monitoring[instance] == BattMonitor_TYPE_ANALOG_VOLTAGE_AND_CURRENT ||
-                _monitoring[instance] == BattMonitor_TYPE_SOLO ||
-                _monitoring[instance] == BattMonitor_TYPE_BEBOP ||
-                _monitoring[instance] == BattMonitor_TYPE_MAXELL);
+    if (instance < _num_instances && drivers[instance] != nullptr && _monitoring[instance] != BattMonitor_TYPE_NONE) {
```

Comment:
Battery driver changes/disabling requires power cycling realistically. SMBus drivers mostly ignore the read function are still updating all the other values (which in turn propagates out to the rest of the system)